lua/lint/linters/markuplint.lua

```diff
@@ -0,0 +1,48 @@
+local binary_name = "markuplint"
+
+local severity_map = {
+  error = vim.diagnostic.severity.ERROR,
+  warning = vim.diagnostic.severity.WARN,
+}
+
+return {
+  cmd = function()
+    local local_binary = vim.fn.fnamemodify("./node_modules/.bin/" .. binary_name, ":p")
+    return vim.loop.fs_stat(local_binary) and local_binary or binary_name
+  end,
+  args = { "--format", "JSON" },
+  stdin = false,
+  stream = "stdout",
+  ignore_exitcode = true,
+  parser = function(output, bufnr)
+    if vim.trim(output) == "" then
+      return {}
+    end
+
+    local decode_opts = { luanil = { object = true, array = true } }
+    local ok, data = pcall(vim.json.decode, output, decode_opts)
+    if not ok then
+      return {
+        {
+          bufnr = bufnr,
+          lnum = 0,
+          col = 0,
+          message = "Could not parse markuplint output due to: " .. data .. "\noutput: " .. output,
+        },
+      }
+    end
```

Comment:
```suggestion
    local data = vim.json.decode(output, decode_opts)
```

This kind of error handling has become default and is no longer necessary per linter.
See https://github.com/mfussenegger/nvim-lint/blob/6f6d866362c74ec2eb036c32b56fac50b02af669/lua/lint/parser.lua#L143-L155